activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -611,21 +611,16 @@ def test_finding_using_primary_key
 
   def test_update_all_on_association_accessed_before_save
     firm = Firm.new(name: "Firm")
-    clients_proxy_id = firm.clients.object_id
     firm.clients << Client.first
     firm.save!
     assert_equal firm.clients.count, firm.clients.update_all(description: "Great!")
-    assert_not_equal clients_proxy_id, firm.clients.object_id
```

Comment:
Will fix it in https://github.com/rails/rails/pull/29098/commits/2c3d99e10a2eed3820606d68aa443516325966d5.